c7n/resources/apigw.py

```diff
@@ -531,9 +531,24 @@ class FilterRestMethod(ValueFilter):
         method={'type': 'string', 'enum': [
             'all', 'ANY', 'PUT', 'GET', "POST",
             "DELETE", "OPTIONS", "HEAD", "PATCH"]},
-        rinherit=ValueFilter.schema)
+        rinherit=ValueFilter.schema, matched={'type': 'boolean'})
     permissions = ('apigateway:GET',)
 
+    def validate(self):
+        if not self.data.get('matched'):
```

Comment:
the validate logic looks like it was copy pasted from somewhere else, can we extract that logic so that it can  be referenced from both places.